### DISCUSSION TITLE
Immediate Charge to Battery

### DISCUSSION THREAD

**Discussion by PatrickTee** [mletenay/home-assistant-goodwe-inverter#240] [Q&A]
Hello everyone. 

Im trying to add a feature to my HA Applikation to instantly load my battery from grid in combination with tibber pricing.
To do so, I need to address the "Immediate Charge to Battery" Function of the converter. I can turn that on by switch in my Solar Go App, but I want to automate that. I tried addressing it via ModBus - but did´nt find a clue. 

Has anyone allready done that or is this easy to add as a feature to the goodwe integration? 

![iScreen Shoter - Safari - 240103123337](https://github.com/mletenay/home-assistant-goodwe-inverter/assets/155535770/031c3aee-cd68-47a0-9500-2e8398d6950f)

**Comment by SnokxStefan** [comment]
have a look here : 
https://github.com/mletenay/home-assistant-goodwe-inverter

**Comment by mletenay** [comment]
It is already part of HA integration since v0.9.9.26. It is called "fast charge" there ...

**Reply by PatrickTee** [reply]
Thank you so much!

**Reply by Ascotg** [reply]
@mletenay I can't seem to find this entity? I have a Goodwe GW10K-ET

**Reply by Ascotg** [reply]
@Ascotg found it, it requires the custom 'experimental' Goodwe integration Goodwe from HACS.
